radar_control/radar_control.py

```diff
@@ -228,315 +285,188 @@ def radar():
     For every pulse, samples and other control information are sent to the n200_driver.
     For every pulse sequence, processing information is sent to the signal processing block.
     After every integration time (AveragingPeriod), the experiment block is given the opportunity
-    to change the control program (if it sends a new one, runradar will halt the old one and begin with
-    the new ControlProg.
+    to change the control program (if it sends a new one, runradar will halt the old one and begin 
+    with the new experiment).
     """
 
-    # Setup socket to send pulse samples over.
-    txsocket=setup_driver_socket()
     # Initialize driverpacket.
-    driverpacket=driverpacket_pb2.DriverPacket()
+    driverpacket = driverpacket_pb2.DriverPacket()
+
+    # Get config options.
+    options = ExperimentOptions()
+
+    context = zmq.Context()  # single context for all sockets involved in this process.
+    # Setup sockets.
+    # Socket to send pulse samples over.
+    try:
```

Comment:
TODO: to test: need to make sure we know that all sockets are set up after this try..except block - test starting programs in different orders